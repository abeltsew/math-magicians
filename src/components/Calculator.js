import React, { useState } from 'react';
import './calculator.css';

import Keys from './Keys';
import calculate from './calculate';

const Calculator = () => {
  const [obj, setObj] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const handleClick = (btn) => {
    setObj(calculate(obj, btn));
  };

  return (
    <div className="calculator">
      <div className="result">
        <p>
          {obj?.total}
          {' '}
          {obj?.operation}
          {' '}
          {obj?.next}
        </p>
      </div>
      {[
        ['AC', '+/-', '%', 'รท'],
        ['7', '8', '9', 'x'],
        ['4', '5', '6', '-'],
        ['1', '2', '3', '+'],
      ].map((row) => (
        <ul key={`row${row[0]}`}>
          {row.map((btn) => (
            <Keys key={btn} btn={btn} clickHandler={() => handleClick(btn)} />
          ))}
        </ul>
      ))}
      <ul>
        <Keys btn="0" cName="zero" clickHandler={() => handleClick('0')} />
        {['.', '='].map((btn) => (
          <Keys key={btn} btn={btn} clickHandler={() => handleClick(btn)} />
        ))}
      </ul>
    </div>
  );
};

export default Calculator;
