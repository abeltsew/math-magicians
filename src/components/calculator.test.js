import renderer from 'react-test-renderer';
import Calculator from './Calculator';

it('should render', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
