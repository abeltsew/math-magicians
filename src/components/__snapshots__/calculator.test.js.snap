// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="calc"
>
  <h1>
    Lets do some Math
  </h1>
  <div
    className="calculator"
  >
    <div
      className="result"
    >
      <p>
        0
         
         
      </p>
    </div>
    <ul
      className="calc-btn"
    >
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="AC"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          AC
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="+/-"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          +/-
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="%"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          %
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="รท"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          รท
        </button>
      </li>
    </ul>
    <ul
      className="calc-btn"
    >
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="7"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          7
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="8"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          8
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="9"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          9
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="x"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          x
        </button>
      </li>
    </ul>
    <ul
      className="calc-btn"
    >
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="4"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          4
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="5"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          5
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="6"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          6
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="-"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          -
        </button>
      </li>
    </ul>
    <ul
      className="calc-btn"
    >
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="1"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          1
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="2"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          2
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="3"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          3
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="+"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          +
        </button>
      </li>
    </ul>
    <ul
      className="calc-btn"
    >
      <li
        className="zero li"
      >
        <button
          className="btn-key"
          id="0"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          0
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="."
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          .
        </button>
      </li>
      <li
        className=" li"
      >
        <button
          className="btn-key"
          id="="
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          =
        </button>
      </li>
    </ul>
  </div>
</div>
`;
